internal/transform/ledger_test.go

```diff
@@ -0,0 +1,153 @@
+	byteFormOfLedgerHeader := []byte(encodedLedgerHeader)
+	parsedTime, err := time.Parse("2006-1-2 15:04:05 MST", "2020-07-12 20:09:07 UTC")
+	desiredOutput := LedgerOutput{
+		Sequence:           int32(sequenceNumber),
+		LedgerHash:         "26932dc4d84b5fabe9ae744cb43ce4c6daccf98c86a991b2a14945b1adac4d59",
+		PreviousLedgerHash: "f63c15d0eaf48afbd751a4c4dfade54a3448053c47c5a71d622668ae0cc2a208",
+		LedgerHeader:       byteFormOfLedgerHeader,
+
+		TransactionCount:           2,
+		OperationCount:             6,
+		SuccessfulTransactionCount: 2,
+		FailedTransactionCount:     201,
+		TxSetOperationCount:        "208",
+
+		ClosedAt: parsedTime,
+
+		TotalCoins:      1054439020873472865,
+		FeePool:         18153766209161,
+		BaseFee:         100,
+		BaseReserve:     5000000,
+		MaxTxSetSize:    1000,
+		ProtocolVersion: 13,
+	}
+	convertedLedger, err := ConvertLedger(testLedger)
+	assert.Equal(t, desiredOutput, convertedLedger)
+
+}
+
+func TestTransformLedger(t *testing.T) {
+	type transformTest struct {
+		input         xdr.LedgerCloseMeta
+		desiredOutput LedgerOutput
+		desiredError  error
+	}
+
+	tests := []transformTest{
+		{
+			wrapLedgerHeader(xdr.LedgerHeader{
+				TotalCoins: -1,
+			}),
+			LedgerOutput{},
+			errors.New("The total number of coins is a negative value"),
+		},
+		{
+			wrapLedgerHeader(xdr.LedgerHeader{
+				FeePool: -1,
+			}),
+			LedgerOutput{},
+			errors.New("The fee pool is a negative value"),
+		},
+		{
+			wrapLedgerHeaderWithTransactions(xdr.LedgerHeader{
+				MaxTxSetSize: 0,
+			}, 2),
+			LedgerOutput{},
+			errors.New("The number of transactions and results are different"),
+		},
+	}
+
+	for _, test := range tests {
+		actualOutput, actualError := ConvertLedger(test.input)
+		assert.Equal(t, test.desiredError, actualError)
+		assert.Equal(t, test.desiredOutput, actualOutput)
+	}
+}
+
+func wrapLedgerHeaderWithTransactions(header xdr.LedgerHeader, numTransactions int) xdr.LedgerCloseMeta {
+	transactionEnvelopes := []xdr.TransactionEnvelope{}
+	for txNum := 0; txNum < numTransactions; txNum++ {
+		transactionEnvelopes = append(transactionEnvelopes, createSampleTx(uint32(txNum)))
+	}
+	lcm, err := xdr.NewLedgerCloseMeta(0, xdr.LedgerCloseMetaV0{
+		LedgerHeader: xdr.LedgerHeaderHistoryEntry{
+			Header: header,
+		},
+		TxSet: xdr.TransactionSet{Txs: transactionEnvelopes},
+	})
+	utils.PanicOnError(err)
+	return lcm
+}
+
+func wrapLedgerHeader(header xdr.LedgerHeader) xdr.LedgerCloseMeta {
+	lcm, err := xdr.NewLedgerCloseMeta(0, xdr.LedgerCloseMetaV0{
+		LedgerHeader: xdr.LedgerHeaderHistoryEntry{
+			Header: header,
+		}})
+	utils.PanicOnError(err)
+	return lcm
+}
+
+func createSampleTx(sequence uint32) xdr.TransactionEnvelope {
+	kp, err := keypair.Random()
+	if err != nil {
```

Comment:
If you're going to create a `PanicOnError` function, may as well use it everywhere :)